src/Kiota.Web/Pages/Generate.razor

```diff
@@ -0,0 +1,221 @@
+    }
+    private async Task GenerateClientInternal() {
+        if(AppInsights != null)
+            await AppInsights.TrackEvent("generateclient", new Dictionary<string, object?> {
+                { "language", Language?.ToString() },
+                { "includePatterns", IncludePatterns },
+                { "excludePatterns", ExcludePatterns },
+                { "additionalData", AdditionalData.ToString() },
+                { "backingStore", BackingStore.ToString() }
+            }).ConfigureAwait(false);
+        var hashedUrl = BitConverter.ToString(HashAlgorithm.Value!.ComputeHash(Encoding.UTF8.GetBytes(DescriptionUrl!.ToString()))).Replace("-", string.Empty);
+        OutputPath = Path.Combine(Path.GetTempPath(), "kiota", "generation", hashedUrl);
+        var generationConfiguration = new GenerationConfiguration{
+            OpenAPIFilePath = DescriptionUrl,
+            IncludePatterns = IncludePatterns?.Split(Environment.NewLine, StringSplitOptions.RemoveEmptyEntries).Select(static x => x.Trim()).ToHashSet(),
+            ExcludePatterns = ExcludePatterns?.Split(Environment.NewLine, StringSplitOptions.RemoveEmptyEntries).Select(static x => x.Trim()).ToHashSet(),
+            Language = Language ?? GenerationLanguage.CSharp,
+            OutputPath = OutputPath,
+            ClientClassName = ClientClassName,
+            ClientNamespaceName = NamespaceName,
+            IncludeAdditionalData = AdditionalData,
+            UsesBackingStore = BackingStore
+        };
+        var logBuilder = new StringBuilder();
+        var builderLogger = new StringBuilderLogger<KiotaBuilder>(LoggerFactory.CreateLogger<KiotaBuilder>(), logBuilder, LogLevel.Warning);
+        var builder = new KiotaBuilder(builderLogger, generationConfiguration);
+        try {
+            await builder.GenerateClientAsync(ComponentDetached).ConfigureAwait(false);
+            var zipFilePath = Path.Combine(Path.GetTempPath(), "kiota", "clients", hashedUrl, "client.zip");
+            if (File.Exists(zipFilePath))
+                File.Delete(zipFilePath);
+            else
+                Directory.CreateDirectory(Path.GetDirectoryName(zipFilePath)!);
+            ZipFile.CreateFromDirectory(OutputPath, zipFilePath);
+            DownloadUrl = zipFilePath;
+            var informationSource = await builder.GetLanguageInformationAsync(ComponentDetached).ConfigureAwait(false) ?? (await GetConfiguration(ComponentDetached).ConfigureAwait(false))?.Languages;
+            if (informationSource?.TryGetValue(generationConfiguration.Language.ToString(), out var languageInformation) ?? false) {
+                foreach(var dependency in languageInformation.Dependencies)
+                    Dependencies += $"{string.Format(languageInformation.DependencyInstallCommand, dependency.Name, dependency.Version)}{Environment.NewLine}";
+            }
+            Logs = logBuilder.ToString();
+        } catch (Exception ex) {
+            Logs += "ERROR" + ex.Message;
```

Comment:
```suggestion
            if(string.IsNullOrEmpty(Logs)) Logs = logBuilder.ToString();
            Logs += "ERROR : " + ex.Message;
```

In case we may want info about which step the error occurred as the exception will most likely prevent line 187 from being executed.